### PR TITLE
modify code related with sync_with_cpp function code  and compile time reduce to less than 10mins

### PR DESCRIPTION
**Pull request by fangshuixun007**
### PR types

Performance optimization
### PR changes

Others
### Describe

1.config as mp=4 sharding=2 pp=32 dp=1 and hack run the compile section
total time 2054s and sync_with_cpp about 70% cost
image
image (1)

2. After modify the sync_with_cpp code, the total time reudce to 481s
image

test machine: yq01-sys-hic-k8s-v100-box-a225-0562
code base : 9e764091f16ff606a5e50a7a998da6cdcd687722

### CODE REVIEW

**Review by gongweibao — DISMISSED** [review]
LGTM

File: python/paddle/fluid/contrib/mixed_precision/fp16_utils.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def _insert_cast_op(block, op, idx, src_dtype, dest_dtype):
                         persistable=False,
                         stop_gradient=in_var.stop_gradient)
 
-                    block._insert_op(
+                    block._insert_op_without_sync(
```
Comment (zhiqiu) [inline python/paddle/fluid/contrib/mixed_precision/fp16_utils.py:119]: Better add a sync before` for loop`?

Comment (fangshuixun007) [reply python/paddle/fluid/contrib/mixed_precision/fp16_utils.py:119]: why add a cast ?
Or you mean add a sync before loop?

**Review by kolinwei — APPROVED** [review]

**Review by zhiqiu — APPROVED** [review]
LGTM for framework.py

**Pull request merged**